decode.js

```diff
@@ -5,34 +5,51 @@ const mri = require('mri')
 const pkg = require('./package.json')
 
 const argv = mri(process.argv.slice(2), {
+    alias: {
+        help: ['h'],
+        version: ['v'],
+    },
 	boolean: [
-		'help', 'h',
-		'version', 'v',
+		'help',
+		'version',
 	]
 })
 
-if (argv.help || argv.h) {
-	process.stdout.write(`
-Usage:
+const usage = `Usage:
     echo -n <url-encoded-string> | url-decode
 Examples:
     echo -n '%7B%22foo%22%3A%20%22bar%22%7D' | url-decode
-    # {"foo": "bar"}
-\n`)
+    # {"foo": "bar"}\n`
+
+if (argv.help) {
+	process.stdout.write(usage)
 	process.exit(0)
 }
 
-if (argv.version || argv.v) {
+if (argv.version) {
 	process.stdout.write(`${pkg.name} v${pkg.version}\n`)
 	process.exit(0)
 }
 
+if (argv._.length > 1) { // Couldn't parse arguments
+    process.stderr.write(usage)
+    process.exit(2)
```

Comment:
I'm curious: Is there a convention (or standard/rule even) that designates `2` for failure to parse arguments?

Reply:
There's no convention. I just picked something other than 1 because having distinct failure exit codes is sometimes nice.